docs/user_guide/basics/flyte_type_system.md

```diff
@@ -0,0 +1,118 @@
+(flyte_type_system_and_data_passing)=
+
+# Flyte type system and Data passing
+
+```{eval-rst}
+.. tags:: Basic
+```
+
+### How It Works
+
+The Flyte type system is designed to ensure that data passed between tasks and workflows adheres to specific formats and types. By leveraging Python’s type hints and type annotations, Flyte can automatically validate inputs and outputs in tasks to ensure correctness.
+
+Flyte supports a wide range of types, including primitive types, collections, and custom types. These are enforced through Flyte’s type engine, which ensures that tasks interact with the correct data structures.
+
+Supported Types:
+
+	1. Primitive Types:
+	    -	int, float, bool, str: Basic data types commonly used in Python functions.
+
+		```{rli} # Define a task to add two integers
+			@task
+			def add(x: int, y: int) -> int:
+    			return x + y
+
+			# Define a task to multiply two integers
+			@task
+			def multiply(x: int, y: int) -> int:
+    		return x * y
+		:caption: basics/task.py
+		:lines: 1
+		```
+
+	2. Collections:
+	    -	List[T]: Represents a list of items where each item is of type T.
+	    -	Dict[K, V]: Represents a dictionary where keys are of type K and values are of type V.
+
+		```{rli} 
+			@task
+			def process_numbers(numbers: List[int]) -> float:
+    			"""Calculate the average of a list of integers."""
+    			return sum(numbers) / len(numbers)
+
+			@task
+			def summarize_data(data: Dict[str, int]) -> str:
+				"""Create a summary of the provided data."""
+				total = sum(data.values())
+				summary = ", ".join([f"{key}: {value}" for key, value in data.items()])
+				return f"Summary: {summary}. Total: {total}"
+		:caption: basics/task.py
+		:lines: 1
+		```
+
+	3. Custom Types:
+	    -	FlyteFile: For handling file inputs and outputs.
+	    -	FlyteSchema: Used for handling structured data (like tables or dataframes).
+	    -	StructuredDataset: Used for handling data in formats like Pandas or Spark DataFrames.
+
+		```{rli} 
+			@task
+			def process_file(file: FlyteFile) -> str:
+    		# Simulate processing the file and returning its name
+    		return f"Processed file: {file}"
+		:caption: basics/task.py
+		:lines: 1
+		```
+
+### When to Use the Type System
```

Comment:
Not sure if this section is needed. Sounds like there are situations when you don't need to use Flyte's type system and that's not the case. This is a core construct in Flyte